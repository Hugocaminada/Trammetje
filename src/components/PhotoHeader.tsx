import React from 'react'
import {Dimensions} from 'react-native'
import styled from 'styled-components/native'
import LinearGradient from 'react-native-linear-gradient'

const windowHeight = Dimensions.get('window').height

const MainContainer = styled.View`
  flex: 1;
`

const StyledGradient = styled(LinearGradient)`
  position: absolute;
  flex: 1;
  z-index: 10;
  height: 100px;
  width: 100%;
`

const HeaderImage = styled.Image`
  width: 100%;
  height: ${windowHeight * 0.25}px;
`

const Content = styled.View`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`

const PhotoHeader = () => (
  <MainContainer>
    <HeaderImage source={require('../assets/header-image.jpg')} />
    <StyledGradient colors={['rgba(0, 0, 0, 0.5)', 'rgba(0, 0, 0, 0)']} />
  </MainContainer>
)

export default PhotoHeader
