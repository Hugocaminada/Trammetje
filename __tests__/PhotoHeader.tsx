/**
 * @format
 */

import 'react-native'
import React from 'react'
import {Text} from 'react-native'
import PhotoHeader from '../src/components/PhotoHeader'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('if it renders correctly', () => {
  const tree = renderer.create(<PhotoHeader />).toJSON()
  expect(tree).toMatchSnapshot()
})
