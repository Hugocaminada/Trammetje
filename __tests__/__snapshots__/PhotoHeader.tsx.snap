// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if it renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/header-image.jpg",
      }
    }
    style={
      Array [
        Object {
          "height": 333.5,
          "width": "100%",
        },
      ]
    }
  />
  <View
    colors={
      Array [
        "rgba(0, 0, 0, 0.5)",
        "rgba(0, 0, 0, 0)",
      ]
    }
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 100,
          "position": "absolute",
          "width": "100%",
          "zIndex": 10,
        },
      ]
    }
  />
</View>
`;
